### PR TITLE
Allow selecting rpc url

### PR DESCRIPTION
**Pull request by supermassive**
Resolves https://github.com/brave/brave-browser/issues/20516

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
Desktop now allows selecting active rpc endpoint. Please also test upgrading from previous versions.
Also test android works as before(no active endpoint feature here and should not get broken)

https://user-images.githubusercontent.com/168356/179480935-9f3f8724-e367-44eb-b12c-5f5ee08f3a25.mp4

### CODE REVIEW

File: components/brave_wallet/common/value_conversion_utils.cc [hunk 1]
```diff
@@ -112,6 +123,78 @@ mojom::NetworkInfoPtr ValueToEthNetworkInfo(const base::Value& value,
...
+  const std::string* chain_id = params_dict->FindString("chainId");
+  if (!chain_id) {
+    return nullptr;
+  }
+  chain.chain_id = *chain_id;
+
+  const std::string* chain_name = params_dict->FindString("chainName");
+  if (chain_name) {
+    chain.chain_name = *chain_name;
+  }
```
Comment (yrliou) [inline components/brave_wallet/common/value_conversion_utils.cc:141]: Have you thought about pulling out common codes between ValueToEthNetworkInfo and ParseEip3085Payload to a  function so we don't need to copy them here and below (Line 174-191)?

**Comment by yrliou** [comment]
Added @SergeyZhukovsky for android review.

**Review by SergeyZhukovsky — APPROVED** [review]
android part++

File: components/brave_wallet/common/value_conversion_utils.cc [hunk 2]
```diff
@@ -244,4 +331,21 @@ base::Value::List PermissionRequestResponseToValue(
   return container_list;
 }
 
+int GetFirstValidChainURLIndex(const std::vector<GURL>& chain_urls) {
```
[the remark this answers is not in the document]
Comment (yrliou) [reply components/brave_wallet/common/value_conversion_utils.cc:334]: resolved via offline communication

File: components/brave_wallet/common/value_conversion_utils.cc (hunk 1 above)
Comment (supermassive) [reply components/brave_wallet/common/value_conversion_utils.cc]: reduced copypaste here https://github.com/brave/brave-core/pull/14104/commits/55c5f85d11113a30791b804f460723aba05beb9f

**Review by yrliou — APPROVED** [review]

**Pull request merged**